docs/technical-reference/Mutation Orchestration Design.md

```diff
@@ -0,0 +1,320 @@
+#### Compilation helpers
+Those engines inject code that are used to reduce the number of compilation errors. Indeed, some mutants alter the control flow which can lead to using non initialized variables, or failing to return a value. So Stryker injects statements to limit the risk of such an event.
+- `AddEndingReturn(...)`: returns a copy of the provided `BlockSyntax` with a `return default(type)` statement added at the end. This method does not add a `return` if:
+	- this is an iteration method (`yield` statement)
+	- return type is `void` or a `Task` (for async method).
+	- the last statement is a `return` or a `throw`
+	- the block does not contain any `return`
+	This method ensures mutations do not break the whole method and trigger roll back logic.
+- `PlaceStaticContextMarker(...)`:  injects the logic used by Stryker to track _static context_ (i.e. code that is run during some static initialization). It relies on a `using` expression. This method supports blocks and expressions.
+	This method must be used to mark any static construct, disregarding mutation.
+- `InjectOutParametersInitialization(...)`: adds statements to initialize to default value any out parameters.
+#### Switching logic injectors
+Those engines inject mutations and the switching logic (`if` or `?: `) to control them.
+-  `PlaceStatementControlledMutations(…)`: build an `if` statement with the provided mutation as the `true` condition and the provided statement/block has the `false` condition. The `if` statement will be chained if there are more than one mutation.
+- `PlaceExpressionControlledMutations`:  build a ternary operator (`?:`) expression with the provided mutation as the `true` condition and the provided expression has the `false` condition . The conditional expression will be chained if there are more than one mutation.
+#### Rollback helpers
+Those engines are used during to remove mutations (and helpers) that cause compilation errors.
+- `RemoveMutant(...)`: returns a syntax node with the mutation removed.
+- `RequiresRemovingChildMutations(...)`: returns true if the syntax node contains an injection that can not be removed without removing any child mutation/injection first.
+- `FindAnnotations(...)`: returns information regarding injection in the provided syntax node, such as used engine and mutant id (if relevant)
+## Node orchestrators
+You can browse this list for figuring out how each syntax structure is mutated, which is useful when trying to understand why Stryker does not give the result you expected or fails.  Remember that Stryker tries out orchestrators in declaration order until one `CanHandle` returns true; so make sure that the most specific orchestrators are declared first.
+Also, you may want to check `DoNotMutateOrchestrator`  when trying to understand why some syntax constructs are not mutated: this may be deliberately.
```

Comment:
```suggestion
Also, you may want to check `DoNotMutateOrchestrator`  when trying to understand why some syntax constructs are not mutated: this may be deliberate.
```